shielder/contract/drink_tests/utils/ops.rs

```diff
@@ -0,0 +1,54 @@
+use drink::AccountId32;
+
+use crate::{contract::OpPub, mocked_zk::ops::OpPriv, types::Scalar};
+
+pub struct UpdateOperation {
+    pub op_pub: OpPub,
+    pub op_priv: OpPriv,
+}
+
+pub fn deposit_op(
+    psp22_address: &AccountId32,
+    user: &AccountId32,
+    amount: u128,
+) -> UpdateOperation {
+    UpdateOperation {
+        op_pub: OpPub::Deposit {
+            amount,
+            token: Scalar {
+                bytes: *((*psp22_address).as_ref()),
+            },
```

Comment:
again, a constructor / casting from `AsRef<[u8; 32]>` will make this code way cleaner

Reply:
Don't agree

I cannot (or I don't know how to) implement

```
impl<T: AsRef<&[u8; 32] >> From<T> for Scalar {
    fn from(value: T) -> Self {
        Self { bytes: *value }
    }
}
```
because it conflicts with `From<u128>` implementatiion which I don't want to remove.

I can do something like

```
impl From<&[u8; 32] > for Scalar {
    fn from(value: &[u8; 32]) -> Self {
        Self { bytes: *value }
    }
}
```

which would require to use
```
let token: Scalar = ((*psp22_address).as_ref() as &[u8; 32]).into();
```
the latter is much more confusing and unreadable rather than explicitly constructing the Scalar from bytes.
I might create `fn new(bytes: [u32; 8]) -> Self` but what is the reason of doing that, if I already have public bytes field? I don't want to encapsulate the `bytes` field.

Reply:
You can have `Scalar::from_bytes` method and call it here.

Reply:
It would be:
```rust
token: Scalar::from_bytes(...)
```
so where would the encapsulation happen?

Reply:
added constructor